firestore/src/common/aggregate_query_snapshot.cc

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "firestore/src/include/firebase/firestore/aggregate_query_snapshot.h"
+
+#include "firestore/src/common/cleanup.h"
+#include "firestore/src/common/util.h"
+#include "firestore/src/include/firebase/firestore/aggregate_query.h"
+
+#if defined(__ANDROID__)
+// TODO(tomandersen) #include
+// "firestore/src/android/aggregate_query_snapshot_android.h"
+#else
+#include "firestore/src/main/aggregate_query_snapshot_main.h"
+#endif  // defined(__ANDROID__)
+
+namespace firebase {
+namespace firestore {
+
+using CleanupFnAggregateQuerySnapshot = CleanupFn<AggregateQuerySnapshot>;
+
+AggregateQuerySnapshot::AggregateQuerySnapshot() {}
+
+AggregateQuerySnapshot::AggregateQuerySnapshot(
+    const AggregateQuerySnapshot& query) {
```

Comment:
Please use `other` for the parameter names of all copy/move constructors and assignment operators, in both the .h and .cc files.

Personally, I like to omit parameter names in the header file in cases where the name does not add any additional value.

For example, instead of

```cpp
AggregateQuerySnapshot(const AggregateQuerySnapshot& other);
```

I like to omit `other` and just use

```cpp
AggregateQuerySnapshot(const AggregateQuerySnapshot&);
```

Then you don't have to worry about keeping the names in sync.

But if you _do_ choose to use a parameter name, they should match in the .h and .cc files. This is something that clang-tidy checks for, and is even enforced in google3: https://clang.llvm.org/extra/clang-tidy/checks/readability/inconsistent-declaration-parameter-name.html.

But most of all, it's best to follow local conventions for parameter names, and other places use `other`, like in [query.h](https://github.com/firebase/firebase-cpp-sdk/blob/b1a5444b10c751c4c039a29bb9fe531b39d1b9ad/firestore/src/include/firebase/firestore/query.h#L90) and [query_snapshot.h](https://github.com/firebase/firebase-cpp-sdk/blob/b1a5444b10c751c4c039a29bb9fe531b39d1b9ad/firestore/src/include/firebase/firestore/query_snapshot.h#L64).

Reply:
I am going to guess that omitting parameter name in header files is a bad idea for public facing APIs. I think there is a direct consequence in generation of documentation. Note the details table that makes reference to parameter name. So if the comments make reference to parameter name, then omission is not an option. 

https://firebase.google.com/docs/reference/cpp/class/firebase/firestore/query-snapshot#classfirebase_1_1firestore_1_1_query_snapshot_1a092feef2ecf7942404beaf56fe7b7b00

I'll make sure header and code files are consistent in my work, but I think we have some cleanup of existing classes if clang-tidy is ever enforced on repo.

Reply:
That's a good point about needing a parameter name in order to document it.